src/test/hostapd_test.rb

```diff
@@ -0,0 +1,296 @@
+    old_driver = Wifi::Hostapd::DEFAULT_OPTIONS[:driver]
+    Wifi::Hostapd::DEFAULT_OPTIONS[:driver] = driver
+
+    Wifi.start
+
+    assert_includes config, "driver=#{driver}"
+  ensure
+    Wifi::Hostapd::DEFAULT_OPTIONS[:driver] = old_driver
+  end
+
+  def test_interface_can_be_changed
+    interface = 'wlan0'
+    old_interface = Wifi::Hostapd::DEFAULT_OPTIONS[:interface_name]
+    Wifi::Hostapd::DEFAULT_OPTIONS[:interface_name] = interface
+
+    Wifi.start
+
+    assert_includes config, "interface=#{interface}"
+  ensure
+    Wifi::Hostapd::DEFAULT_OPTIONS[:interface_name] = old_interface
+  end
+
+  def test_hw_capabilities_are_read_from_iw_info
+    # see fixtures/bin/iw-info.out
+    Wifi.start
+    assert_includes config, "ht_capab=[HT20][HT40+][SHORT-GI-20][SHORT-GI-40][DSSS_CCK-40][TX-STBC][RX-STBC1]"
+  end
+
+  def test_ieee80211n_set_true_from_iw_info
+    # see fixtures/bin/iw-info-true.out
+    ENV['MOCK_IW_80211N'] = "TRUE"
+    Wifi.start
+    assert_includes config, "ieee80211n=1"
+  end
+
+  def test_ieee80211n_set_false_from_iw_info
+    ENV['MOCK_IW_80211N'] = "FALSE"
+    # see fixtures/bin/iw-info-false.out
+    Wifi.start
+    assert_includes config, "ieee80211n=0"
+  ensure
+    ENV.delete('MOCK_IW_80211N')
+  end
+
+  def test_wmm_enabled_set_true
+    # this is just the default switch, advanced config
+    # depends on hardware features
+    Wifi.start
+    assert_includes config, "wmm_enabled=1"
+  end
+
+  def test_wme_enabled_set_true
+    # no idea what this does, maybe disable it?
+    Wifi.start
+    assert_includes config, "wmm_enabled=1"
+  end
+
+
+  def test_hwmode_set_to_g
+    # we can use #11b, 11g, and 11a respectively:
+    # b is slow, a is 5GHz only and g the only sane solution
+    # (ng and na are no valid options, see https://dev.openwrt.org/ticket/17541)
+    Wifi.start
+    assert_includes config, "hw_mode=g"
+  end
+
+  def test_ieee80211d_set_true
+    # must be enabled to a) announce power and channel settings and
+    # b) to enable ieee80211d (RADAR detection and DFS support)
+    Wifi.start
+    assert_includes config, "ieee80211d=1"
+  end
+
+  def test_ieee80211h_set_true
+    # enables RADAR detection and DFS support
+    Wifi.start
+    assert_includes config, "ieee80211h=0"
+  end
+
+  def test_ht40_is_positive_when_channel_is_smaller_than_8
+    1.upto 7 do |channel|
+      with_channel(channel) { Wifi.start }
+      assert_includes config, "ht_capab=[HT20][HT40+][SHORT-GI-20][SHORT-GI-40][DSSS_CCK-40][TX-STBC][RX-STBC1]"
+    end
+  end
+
+  def test_ht40_is_negative_when_channel_is_greater_than_7
+    8.upto 15 do |channel|
+      with_channel(channel) { Wifi.start }
+      assert_includes config, "ht_capab=[HT20][HT40-][SHORT-GI-20][SHORT-GI-40][DSSS_CCK-40][TX-STBC][RX-STBC1]"
+    end
+  end
+
+  ["private", "public"].each do |section|
+    define_method("test_#{section}_macaddr_acl_is_0") do
+      self.method("#{section}_disabled").call { Wifi.start }
```

Comment:
Hmm: `in 'test': wrong number of arguments (given 1, expected 2) (ArgumentError)` @colszowka